GVFS/GVFS.Common/ProductUpgrader.Shared.cs

```diff
@@ -19,12 +18,25 @@ public static bool IsLocalUpgradeAvailable()
             string downloadDirectory = GetAssetDownloadsPath();
             if (Directory.Exists(downloadDirectory))
             {
-                const string PotentialInstallerName = "*VFS*.*";
-                string[] installers = Directory.GetFiles(
-                    downloadDirectory,
-                    PotentialInstallerName, 
-                    SearchOption.TopDirectoryOnly);
-                return installers.Length > 0;
+                // This method is used Only by Git hooks. Git hooks does not have access
+                // to GVFSPlatform to read platform specific file extensions. That is the
+                // reason possible installer file extensions are defined here.
+                HashSet<string> extensions = new HashSet<string>() { "EXE", "DMG", "RPM", "DEB" };
+                HashSet<string> installerNames = new HashSet<string>()
+                {
+                    GVFSInstallerFileNamePrefix.ToUpperInvariant(),

```

Comment:
You don't need to change any of these strings. Just use case-insensitive lookups in the HashSet.